### DISCUSSION TITLE
Diffusion problem cannot converge with small scale model

### DISCUSSION THREAD

**Discussion by Chaoyue-Liu** [idaholab/moose#22397] [Q&A General] [answered]
Hello,

I'm trying to do a simple transient diffusion calculation within a small domain with size of 50*50 µm. It somehow cannot converge properly. The input file is here: [https://github.com/Chaoyue-Liu/battery/blob/main/problem/simplediffusion.i](url)

I also tried to calculate the same problem at steady state, by ignoring TimeDerivative kernel and adding a DirichletBC condition. The calculation works well ([https://github.com/Chaoyue-Liu/battery/blob/main/problem/steady_diffuison.i](url)), but if I calculate this in transient, it doesn't converge [https://github.com/Chaoyue-Liu/battery/blob/main/problem/Transient_diffusion.i](url) . Could you check what am I doing wrong? Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

The residuals are very small on this case, because you are very close to the solution with your 0 initialization it seems. 
You can raise the tolerances on the linear and nonlinear solver when the solver seems to never accept the solution, even though the residual is very small. 

This input converges fine 
```

    [Mesh]
        type = GeneratedMesh
        dim = 2
        nx = 50
        ny = 50
        xmax = 0.00005
        ymax = 0.00005
    []
    
    [Variables]
    [phis]
    []
    []
    
    [Materials]
    [d]
    type=GenericConstantMaterial
    prop_names='sigmas'
    prop_values='1'
    []
    []
    
    [Kernels]
    [ohm_s]
    type=MatDiffusion
    variable=phis
    diffusivity='sigmas'
    []
    [time_s]
    type=TimeDerivative
    variable=phis
    []
    []
    
    [ICs]
    [phis_ic]
    type=ConstantIC
    variable=phis
    value=0
    []
    []
    
    [BCs]
    [phis_neumann]
    boundary='bottom'
    type=ADNeumannBC
    variable=phis
    value=0.0001
    []
    [phis_dirichlet]
    boundary='top'
    type=ADDirichletBC
    variable=phis
    value=0
    []
    []
    
    [Executioner]
    type=Transient
    dt=1
    end_time=100
    automatic_scaling=true
    nl_abs_tol = 1e-12
    l_tol = 1e-8
    []
    
    [Outputs]
    exodus=true
    []
```


PS: the links are not set up properly, you need to do this : `[text for the link](address for the link)`

**Reply by Chaoyue-Liu** [reply]
Thank you. It is indeed the problem with tolerance. I increased it and it converges fine. Thanks for the guidance.
